kura/examples/org.eclipse.kura.example.ibeacon.advertiser/src/main/java/org/eclipse/kura/example/ibeacon/advertiser/IBeaconAdvertiserOptions.java

```diff
@@ -43,20 +53,25 @@
     private final Integer txPower;
     private final String iname;
 
+    private static final Logger logger = LoggerFactory.getLogger(IBeaconAdvertiserOptions.class);
+
     public IBeaconAdvertiserOptions(Map<String, Object> properties) {
         requireNonNull(properties, "Required not null");
         this.enable = getProperty(properties, PROPERTY_ENABLE, PROPERTY_ENABLE_DEFAULT);
         this.minInterval = (int) (getProperty(properties, PROPERTY_MIN_INTERVAL, PROPERTY_MIN_INTERVAL_DEFAULT)
                 / 0.625);
         this.maxInterval = (int) (getProperty(properties, PROPERTY_MAX_INTERVAL, PROPERTY_MAX_INTERVAL_DEFAULT)
                 / 0.625);
-        this.major = getProperty(properties, PROPERTY_MAJOR, PROPERTY_MAJOR_DEFAULT);
-        this.minor = getProperty(properties, PROPERTY_MINOR, PROPERTY_MINOR_DEFAULT);
-        this.txPower = getProperty(properties, PROPERTY_TX_POWER, PROPERTY_TX_POWER_DEFAULT);
+        this.major = checkRange(getProperty(properties, PROPERTY_MAJOR, PROPERTY_MAJOR_DEFAULT), PROPERTY_MAJOR_MAX,
+                PROPERTY_MAJOR_MIN);
+        this.minor = checkRange(getProperty(properties, PROPERTY_MINOR, PROPERTY_MINOR_DEFAULT), PROPERTY_MINOR_MAX,
+                PROPERTY_MINOR_MIN);
+        this.txPower = checkRange(getProperty(properties, PROPERTY_TX_POWER, PROPERTY_TX_POWER_DEFAULT),
+                PROPERTY_TX_POWER_MAX, PROPERTY_TX_POWER_MIN);
         this.iname = getProperty(properties, PROPERTY_INAME, PROPERTY_INAME_DEFAULT);
-
         String uuidString = getProperty(properties, PROPERTY_UUID, PROPERTY_UUID_DEFAULT);
         if (uuidString.trim().replace("-", "").length() != 32) {
+            logger.warn("UUID is too short or too long!");
```

Comment:
Is the uuid hex check missing?

Reply:
Yes it seems it is. I will added it right away. Thank you for your input on it.